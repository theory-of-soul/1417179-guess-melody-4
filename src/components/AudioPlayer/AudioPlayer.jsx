import React from "react";
import PropTypes from "prop-types";

const AudioPlayer = (props) => {
  return (
    <>
      <button
        className={`track__button track__button--${props.isPlaying ? `pause` : `play`}`}
        type="button"
        disabled={props.isLoading}
        onClick={props.handlePlay}
      />
      <div className="track__status">
        {props.children}
      </div>
    </>
  );
};

AudioPlayer.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  handlePlay: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};

export default AudioPlayer;
