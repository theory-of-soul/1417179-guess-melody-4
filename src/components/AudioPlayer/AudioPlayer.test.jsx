import React from "react";
import renderer from 'react-test-renderer';
import AudioPlayer from "./AudioPlayer";

const src = ``;
const options = {
  createNodeMock: (element) => {
    if (element.type === `audio`) {
      return {
        oncanplaythrough() {},
      };
    }
    return null;
  }
};

const player = <audio src={src}/>;

describe(`AudioPlayer snapshot tests`, () => {
  it(`AudioPlayer is silent and loading`, () => {
    const component = renderer
      .create(
          <AudioPlayer
            isPlaying={false}
            isLoading={true}
            handlePlay={() => {}}
          >
            {player}
          </AudioPlayer>,
          options
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it(`AudioPlayer is playing`, () => {
    const component = renderer
      .create(
          <AudioPlayer
            isPlaying={true}
            isLoading={false}
            handlePlay={() => {}}
          >
            {player}
          </AudioPlayer>,
          options
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
