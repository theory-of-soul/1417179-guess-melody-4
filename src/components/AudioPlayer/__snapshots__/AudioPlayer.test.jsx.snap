// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer snapshot tests AudioPlayer turn off state 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      src=""
    />
  </div>,
]
`;

exports[`AudioPlayer snapshot tests AudioPlayer turn on state 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      src=""
    />
  </div>,
]
`;
